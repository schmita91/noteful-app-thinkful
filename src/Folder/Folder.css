.delete-folder {
    border-radius: 5px;
    width: 60px;
    display: block;
    background-color: #096488;
    border: 1px solid #0B99D3;
    color: #F0F7FA;
    margin-left: 10px;


}

.folder {
    height: 50px;
    border: 1px solid black;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0px 10px;
    background-color: #064c68;
}

/* FOLDER HIGHLIGHTER */
.active {
    background:#0B99D3;
}
.highlighted {
    background:rgb(87, 190, 231);
    color: #F0F7FA;
 }