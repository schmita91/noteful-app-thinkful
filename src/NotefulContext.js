import React from 'react'

const NotefulContext = React.createContext({
    folders:[],
    notes:[],
    addFolder: ()=>{},
    deleteNote: ()=>{},
    deleteFolder: ()=>{},
    addNote: ()=>{}
})

export default NotefulContext;