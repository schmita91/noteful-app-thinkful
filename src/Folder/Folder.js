import React, {Component} from 'react';
import './Folder.css'
import {NavLink} from 'react-router-dom';
import NotefulContext from '../NotefulContext';
import PropTypes from 'prop-types';
import config from '../config'


class Folder extends Component {
    static contextType = NotefulContext;
    handleDeleteFolder(id, cb, routerProps){
        fetch( `${config.API_ENDPOINT}/api/folders/${id}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json'
            },
        })
        .then (res => {
            if (!res.ok){
                return res.json().then(error=> {
                    throw error
                })
            }
        })
        .then(() => {
            cb(id);
    
        })
        .then(()=>{
            routerProps.routerProps.push('/')
        })
        .catch(error=> {
            alert('something went wrong, please try again')
        })
    }
    render(){
        const folderNum = Number(this.props.routerProps.folderid)
        const folderToHighlight = this.context.folders.find(folder=> folder.id === folderNum) || {};
        return (
            <NotefulContext.Consumer>
                {(context)=> {
                return (
                    <NavLink to={this.props.link}>
                    <li key={this.props.folderid}
                    className={this.props.folderid === folderToHighlight.id ? 'folder highlighted' : 'folder'}>
                    {this.props.name}
                    <button className="delete-folder" id={this.props.id} onClick={()=>this.handleDeleteFolder(this.props.folderid, context.deleteFolder, this.props.routerProps)}>Delete</button>
                    </li>
                </NavLink>
                )
                }
            }
                </NotefulContext.Consumer>
        )
    }
}
Folder.propTypes = {
    folderid: PropTypes.number.isRequired,
    link: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  };
export default Folder;