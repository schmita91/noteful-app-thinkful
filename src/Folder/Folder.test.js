import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Folder from './Folder'
import NotefulContext from '../NotefulContext';
import {BrowserRouter} from 'react-router-dom'

describe('folder component', ()=> {
  
const routerp =  {"folderid": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
"name": "Important"}

const folders = [
    {
      "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Important"
    },
    {
      "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Super"
    },
    {
      "id": "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Spangley"
    }
  ];

it('renders to UI as expected', ()=> {

const tree = renderer.create(
  <BrowserRouter>
<NotefulContext.Provider value={{folders}}>
    <Folder routerProps={routerp} folderid="b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1" link='link' />
</NotefulContext.Provider>
</BrowserRouter>)
.toJSON()
expect(tree).toMatchSnapshot();
})
})
