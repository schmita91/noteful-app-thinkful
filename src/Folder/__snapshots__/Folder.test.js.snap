// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`folder component renders to UI as expected 1`] = `
<a
  aria-current={null}
  href="/link"
  onClick={[Function]}
>
  <li
    className="folder"
  >
    <button
      className="delete-folder"
      onClick={[Function]}
    >
      Delete
    </button>
  </li>
</a>
`;
